src/iface/ethernet.rs

```diff
@@ -761,11 +764,63 @@ impl<'b, 'c> InterfaceInner<'b, 'c> {
             // Ignore any echo replies.
             Icmpv6Repr::EchoReply { .. } => Ok(Packet::None),
 
+            // Forward any NDISC packets to the ndisc packet handler
+            Icmpv6Repr::Ndisc(repr) if ip_repr.hop_limit() == 0xff => match ip_repr {
+                IpRepr::Ipv6(ipv6_repr) => self.process_ndisc(timestamp, ipv6_repr, repr),
+                _ => Ok(Packet::None)
+            },
+
             // FIXME: do something correct here?
             _ => Err(Error::Unrecognized),
         }
     }
 
+    #[cfg(feature = "proto-ipv6")]
+    fn process_ndisc<'frame>(&mut self, timestamp: Instant, ip_repr: Ipv6Repr,
+                             repr: NdiscRepr<'frame>) -> Result<Packet<'frame>> {
+        let packet = match repr {
+            // TODO: Use the neighbor advert flags and handle the advertisement acordingly
+            NdiscRepr::NeighborAdvert { lladdr, target_addr, .. } => {
+                match lladdr {
+                    Some(lladdr) if lladdr.is_unicast() && target_addr.is_unicast() => {
+                        self.neighbor_cache.fill(ip_repr.src_addr.into(), lladdr, timestamp)
+                    },
+                    _ => (),
+                }
+                Ok(Packet::None)
+            }
+            NdiscRepr::NeighborSolicit { target_addr, lladdr, .. } => {
+                match lladdr {
+                    Some(lladdr) if lladdr.is_unicast() && target_addr.is_unicast() => {
+                        self.neighbor_cache.fill(ip_repr.src_addr.into(), lladdr, timestamp)
+                    },
+                    _ => (),
+                }
+                if self.has_solicited_node(ip_repr.dst_addr) && self.has_ip_addr(target_addr) {
+                    let advert = Icmpv6Repr::Ndisc(NdiscRepr::NeighborAdvert {
+                        flags: NdiscNeighborFlags::SOLICITED,
+                        target_addr: target_addr,
+                        //target_addr: ip_repr.src_addr,
+                        lladdr: Some(self.ethernet_addr)
+                    });
+                    let ip_repr = Ipv6Repr {
+                        src_addr: target_addr,
+                        dst_addr: ip_repr.src_addr,
+                        next_header: IpProtocol::Icmpv6,
+                        hop_limit: 0xff,
+                        payload_len: advert.buffer_len()
+                    };
+                    Ok(Packet::Icmpv6((ip_repr, advert)))
+                } else {
+                    Ok(Packet::None)
+                }
+            }
+            // TODO: Handle more types of NDISC packets
```

Comment:
Which packet types we don't handle, but should?

Reply:
RouterAdvertisements, RouterSolicitations, and Redirect messages aren't currently handled.

Reply:
Can you add the current status of NDISC support to README in the usual format instead of this TODO?

Reply:
Added some sections to the README